homeassistant/components/tfa_me/config_flow.py

```diff
@@ -0,0 +1,177 @@
+    SelectSelectorConfig,
+    SelectSelectorMode,
+)
+
+from .const import CONF_NAME_WITH_STATION_ID, DEFAULT_STATION_NAME, DOMAIN
+from .coordinator import TFAmeDataCoordinator
+from .data import TFAmeData, TFAmeException
+
+# Scheme for IP/Domain and poll interval
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_IP_ADDRESS): str,
+        vol.Required(CONF_NAME_WITH_STATION_ID): bool,
+    }
+)
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TFAmeConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle the config flow for TFA.me stations."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self.data: dict[str, Any] = {}
+        self.name_with_station_id: bool = False
+
+    _LOGGER.debug("TFA.me config flow")
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """First step: Configuration UI."""
+        errors: dict[str, str] = {}
+
+        # No input (empty)
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=DATA_SCHEMA, errors=errors
+            )
+
+        # Get multiple entities option
+        multi_ent = user_input.get(CONF_NAME_WITH_STATION_ID)
+        if not isinstance(multi_ent, bool):
+            self.name_with_station_id = False
+            errors[CONF_NAME_WITH_STATION_ID] = "invalid_name_with_station_id"
+            # Error, multiple entities validation failed
+            return self.async_show_form(
+                step_id="user", data_schema=DATA_SCHEMA, errors=errors
+            )
+        self.name_with_station_id = multi_ent
+
+        # Get IP or mDNS host name
+        ip_host_str = user_input.get(CONF_IP_ADDRESS)
+
+        # If user_input is not None:
+        validator = TFAmeValidator()
+        if validator.is_valid_ip_or_tfa_me(ip_host_str):
+            title_str: str = DEFAULT_STATION_NAME
+            if isinstance(ip_host_str, str):
+                title_str = DEFAULT_STATION_NAME + " '" + ip_host_str.upper() + "'"
+
+            try:
+                client = TFAmeData(user_input[CONF_IP_ADDRESS])
+                identifier = await client.get_identifier()
+            except TFAmeException:
+                errors["base"] = "host_empty"
+            except Exception:
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                await self.async_set_unique_id(identifier)
+                self._abort_if_unique_id_configured()
+
+                # Create a TFA.me device entry
+                return self.async_create_entry(title=title_str, data=user_input)
+
+        # Update error list: invalid IP or host
+        errors[CONF_IP_ADDRESS] = "invalid_ip_host"
+
+        # Error, validation failed
+        return self.async_show_form(
+            step_id="user", data_schema=DATA_SCHEMA, errors=errors
+        )
```

Comment:
I think this should be rewritten, you can tidy this up and remove duplicate function calls by opposing the logic on `user_input` (rather than checking if user_input is None, check that it is not None and perform all your validation / setup underneath.

Then the last piece of logic in the chain will always be to show the form.